File: Lab2/MyString.cpp
#include "MyString.h"
#include <iostream>

MyString::MyString() : str("") {}
MyString::MyString( int* s) {}
MyString::MyString( string& s) : str(s) {}
MyString::MyString(MyString& other) : str(other.str) {}
bool MyString::Check() {
    for (char c : str) {
        if (!isdigit(c)) {
            return false;
        }
    }
    return true;
}
int MyString::length() {
    return str.length();
}
void MyString::removeFive() {
    size_t pos = str.find("5");
    while (pos != string::npos) {
        str.erase(pos, 1);
        pos = str.find("5");
    }
}
int MyString::length2() {
    return str.length();
}
void MyString::print() {
    cout << str << endl;
}
void MyString::Add(string add){
    str += add;
}

File: Lab2.2/MyString.cs
using System;

namespace MyStringNamespace{
    class MyString{
        private string str; // Значення рядка
        public MyString(){ // Конструктор за замовчуванням              
            str = "";
        }
        public MyString(int[] s){ // Конструктор зі значенням C-style рядка
            str = string.Join("", Array.ConvertAll(s, i => i.ToString()));
        }
        public MyString(string s){ // Конструктор зі значенням рядка типу string
            str = s;
        }
        public MyString(MyString other){ // Конструктор копіювання
            str = other.str;
        }
        public bool Check()
        { // Перевірка елементів
            foreach (char c in str)
            {
                if (!Char.IsDigit(c))
                {
                    return false;
                }
            }
            return true;
        }
        public int Length(){ // Метод обчислення довжини рядка
            return str.Length;
        }
        public int Length2(){ // Метод обчислення довжини рядка після видалення
            return str.Length;
        }

        public void RemoveFive(){ // Метод видалення символу '5'
            str = str.Replace("5", "");
        }
        public void Print(){ // Метод виведення рядка на екран
            Console.WriteLine(str);
        }
        public void Add(string a){ // Метод додавання символу
            str += a;
        }
    }
}

